### PR TITLE
Remove mentions of C from the docs and document the use of @ in grammar files

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**
I also took the liberty to add support for @class that is showcased in examples but was not actually supported.

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
A quick question, some examples make use of the syntax `rule[type] (memo):` but the python parser generator does not seem to make use of that hint ? Should it ? It looks like everything is memoized anyway so I am not sure what is the purpose of this.

File: docs/grammar.md [hunk 1]
```diff
@@ -111,13 +110,7 @@ rule_name[return_type]:
...
     | second_alt1 second_alt2 { second_alt1 }
 ```
-If the action is omitted and C code is being generated, then there
-are two different possibilities:
-1.  If there's a single name in the alternative, this gets returned.
-2.  If not, a dummy name object gets returned.
-
-If Python code is being generated, then a list with all the parsed
-expressions gets returned.
+If the action is omitted, a list with all the parsed expressions gets returned.
```
Comment (lysnikolaou) [inline docs/grammar.md:113]: This should probably be changed after #14, right?

**Review by lysnikolaou — COMMENTED** [review]
Thanks a lot for the PR, @MatthieuDartiailh! I've left one comment.

File: docs/grammar.md (hunk 1 above)
Comment (MatthieuDartiailh) [reply docs/grammar.md:113]: Indeed

Comment (MatthieuDartiailh) [reply docs/grammar.md:113]: Addressed.

File: docs/grammar.md [hunk 2]
```diff
@@ -128,6 +124,51 @@ The name can then be used in the action, like this:
 rule_name[return_type]: '(' a=some_other_rule ')' { a }
 ```
 
+
+### Customizing the generated parser
+
+By default, the generated parser inherits from the Parser class defined in pgen
+Parser, and is named GeneratedParser. One can customize the generated module by
```
Comment (lysnikolaou) [inline docs/grammar.md:131]:
```suggestion
By default, the generated parser inherits from the Parser class defined in pegen/parser.py,
and is named GeneratedParser. One can customize the generated module by
```

File: docs/grammar.md [hunk 3]
```diff
@@ -128,6 +124,51 @@ The name can then be used in the action, like this:
...
+modifying the header and trailer of the module generated by pegen. To do so one
+can add to the grammar dedicated sections discussed below:
```
Comment (lysnikolaou) [inline docs/grammar.md:133]:
```suggestion
can add dedicated sections to the grammar, which are discussed below:
```

File: docs/grammar.md [hunk 4]
```diff
@@ -128,6 +124,51 @@ The name can then be used in the action, like this:
...
+
+@class NAME
+
+This allows to specify the name of the generated parser.
+
+@header
+
+  Specify the header of the module as a string (one can typically use triple
```
Comment (lysnikolaou) [inline docs/grammar.md:141]: The indentation here is not consistent with the other `meta`s. Let's have it consistent everywhere.

File: docs/grammar.md (hunk 1 above)
Comment (lysnikolaou) [reply docs/grammar.md:113]: Thanks.

File: docs/grammar.md (hunk 4 above)
Comment (MatthieuDartiailh) [reply docs/grammar.md:141]: I went with 2 space which makes for a nicer reading of the source IMO but does not cause the markdown to interprete it as code.

**Review by lysnikolaou — APPROVED** [review]
Thanks again @MatthieuDartiailh!

**Comment by lysnikolaou** [comment]
@pablogsal I reviewed this, but leaving it open for you to have another look, since you assigned it to yourself.

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
Sorry for the delay @MatthieuDartiailh  and thanks a lot for the PR! 🎉

**Pull request merged**